File: frontend/src/types/Project.ts
import { Task } from './index';
import { User } from './User';

interface TeamMember {
    id: number;
    email: string;
    full_name: string;
}

interface Team {
    id: number;
    name: string;
    members: TeamMember[];
}

export interface Project {
    id: number;
    name: string;
    team_id: number | null;
    created_at: string;
    manager_id: number;
    manager: User;
    team?: Team;
    tasks: Task[];
} 

File: backend/src/models/project.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from src.database.config import Base

class Project(Base):
    __tablename__ = "projects"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False)
    team_id = Column(Integer, ForeignKey('teams.id', ondelete="CASCADE"), nullable=True)
    created_at = Column(DateTime, default=datetime.utcnow)
    manager_id = Column(Integer, ForeignKey('users.id', ondelete="SET NULL"), nullable=False)

    # Relationships
    team = relationship("Team", back_populates="projects")
    tasks = relationship("Task", back_populates="project", cascade="all, delete-orphan")
    manager = relationship("User", foreign_keys=[manager_id], back_populates="managed_projects") 